Add chart to the generated Excel file

SofiBecerril opened the pull request.

SofiBecerril committed:
refactor: :adhesive_bandage: Code to create the directory "reports"

SofiBecerril committed:
refactor: :adhesive_bandage: Code to create the directory "reports"

SofiBecerril committed:
refactor: :adhesive_bandage: Code to create the directory "reports"

SofiBecerril committed:
refactor: :adhesive_bandage: Code to create the directory "reports"

SofiBecerril committed:
refactor: :adhesive_bandage: Code to create the directory "reports"

SofiBecerril committed:
refactor: :adhesive_bandage: Code to create the directory "reports"

SofiBecerril committed:
refactor: :adhesive_bandage: Correction in report

SofiBecerril committed:
refactor: :adhesive_bandage: Correction in report

SofiBecerril committed:
refactor: :adhesive_bandage: Correction in report

SofiBecerril committed:
refactor: :adhesive_bandage: Correction in report

SofiBecerril committed:
refactor: :adhesive_bandage: Correction in report

SofiBecerril committed:
Fix PHP styling 🎨

SofiBecerril committed:
Fix PHP styling 🎨

SofiBecerril committed:
Fix PHP styling 🎨

SofiBecerril committed:
Fix PHP styling 🎨

SofiBecerril committed:
Fix PHP styling 🎨

EliBece committed:
refactor: :art: Adapt code and add send icon

Se hicieron cambios en el código referente a SED para adaptarlo a las nuevas funciones de Database. Se agregó a la interfaz de SED un icono para enviar emails.

EliBece committed:
refactor: :art: Adapt code and add send icon

Se hicieron cambios en el código referente a SED para adaptarlo a las nuevas funciones de Database. Se agregó a la interfaz de SED un icono para enviar emails.

EliBece committed:
Fix PHP styling

EliBece committed:
Fix PHP styling

EliBece committed:
Merge branch 'dev-eli' of https://github.com/OnlyAlec/ULSA_ProcesosPosgrado into dev-eli

EliBece committed:
Fix Prettier styling 🎨

OnlyAlec committed:
feat: :sparkles: Install node dependencies

- Brevo

EliBece committed:
feat: :sparkles: For SED send emails and minor changes.

Se agrego la posibilidad de enviar emails a los alumnos que aún no han realizado la evaluación docente. Se realizaron pequeños cambios en el ccódigo.

EliBece committed:
Fix Prettier styling 🎨

OnlyAlec committed:
fix(Brevo): :bug: Fix CA path and MJML construct

OnlyAlec committed:
style: :art: Improvements

OnlyAlec committed:
style: :lipstick: Update UI using the Look&Feel

OnlyAlec committed:
Fix PHP styling 🎨

EliBece committed:
refactor: :recycle: Changes in Class Mailer.

Se hicieron modificaciones a la Clase Mailer. Se eliminó el archivo "Confirmation" de SED.

EliBece committed:
Merge branch 'dev-eli' of https://github.com/OnlyAlec/ULSA_ProcesosPosgrado into dev-eli

EliBece committed:
Fix PHP styling 🎨

OnlyAlec committed:
perf: :goal_net: Standarize catching errors of AFI

OnlyAlec committed:
perf: :goal_net: Standarize catching errors other modules

OnlyAlec committed:
fix: :sparkles: Making `mailer` more modular

OnlyAlec committed:
fix: :bug: Update mailer and  gitignore

SofiBecerril committed:
feat: :sparkles: Add charts to the generated Excel file

SofiBecerril committed:
feat: :sparkles: Add charts to the generated Excel file

SofiBecerril committed:
feat: :sparkles: Add charts to the generated Excel file

SofiBecerril committed:
feat: :sparkles: Add charts to the generated Excel file

SofiBecerril committed:
feat: :sparkles: Add charts to the generated Excel file

SofiBecerril committed:
Fix PHP styling 🎨

SofiBecerril committed:
Fix PHP styling 🎨

SofiBecerril committed:
Fix PHP styling 🎨

SofiBecerril committed:
Fix PHP styling 🎨

SofiBecerril committed:
Fix PHP styling 🎨

OnlyAlec reviewed (changes requested):
Señorita Sofi, perfecto trabajo. Existe nada mas un pequeño detalle 👍🏻

omarrdbz committed:
wip

omarrdbz committed:
Fix Prettier styling 🎨

omarrdbz committed:
look&feel GA

omarrdbz committed:
Merge branch 'dev-omar' of https://github.com/OnlyAlec/ULSA_ProcesosPosgrado into dev-omar

omarrdbz committed:
Merge remote-tracking branch 'origin/dev-alec' into dev-omar

omarrdbz committed:
minor changes

omarrdbz committed:
Merge remote-tracking branch 'origin/dev-eli' into dev-omar

omarrdbz committed:
Merge remote-tracking branch 'origin/dev-sofi' into dev-omar

omarrdbz committed:
look&feel

omarrdbz committed:
Fix Prettier styling 🎨

omarrdbz committed:
minor changes

omarrdbz committed:
Merge branch 'dev-omar' of https://github.com/OnlyAlec/ULSA_ProcesosPosgrado into dev-omar

omarrdbz committed:
Fix PHP styling 🎨

OnlyAlec committed:
fix(Email): :bug: Create email templated and fix `Mailer`

omarrdbz committed:
need fix: datalist filter SED

omarrdbz committed:
Merge branch 'dev-omar' of https://github.com/OnlyAlec/ULSA_ProcesosPosgrado into dev-omar

omarrdbz committed:
Fix Prettier styling 🎨

OnlyAlec committed:
fix: 🐛 Minor normalization

OnlyAlec committed:
fix: :bug: Fix event selectors and UI

OnlyAlec committed:
Fix Prettier styling 🎨

OnlyAlec committed:
Merge pull request #93 from OnlyAlec/dev-alec

[FIX] ✨ Deploy of v2.1.0 AFI module

OnlyAlec committed:
Merge branch 'development' into dev-eli

OnlyAlec committed:
Fix PHP styling 🎨

OnlyAlec committed:
Merge pull request #91 from OnlyAlec/dev-eli

Modulo SED: Enviar emails y cambios menores.

OnlyAlec committed:
fix: 🐛 Adding dependencies

OnlyAlec committed:
refactor: :lipstick: Use same style in emails

omarrdbz committed:
fix: SED update_from_excel

omarrdbz committed:
Merge branch 'dev-omar' of https://github.com/OnlyAlec/ULSA_ProcesosPosgrado into dev-omar

omarrdbz committed:
Fix PHP styling 🎨

SofiBecerril committed:
Merge branch 'dev-sofi' of https://github.com/OnlyAlec/ULSA_ProcesosPosgrado into dev-sofi

OnlyAlec committed:
fix: :ambulance: Fixed programs displayed

OnlyAlec committed:
style: :lipstick: Display with capitalized name

omarrdbz committed:
fix: look&feel in SED

omarrdbz committed:
Merge branch 'dev-omar' of https://github.com/OnlyAlec/ULSA_ProcesosPosgrado into dev-omar

omarrdbz committed:
Fix Prettier styling 🎨

OnlyAlec committed:
Merge pull request #95 from OnlyAlec/dev-alec

[FIX] Hotfix for AFI and Email standard

OnlyAlec commented on public/modules/AFI/formsDB.php:
Se va a hacer un cambio de librería, para mejor control

OnlyAlec committed:
Merge branch 'development' into dev-omar

Signed-off-by: Alexis Chacon <66844946+OnlyAlec@users.noreply.github.com>

OnlyAlec committed:
fix: :bug: Minor fix

OnlyAlec committed:
Merge pull request #94 from OnlyAlec/dev-omar

[UI] Look&Feel in modules GA/SED

SofiBecerril committed:
feat: :sparkles: Generate master's and specialities charts using Node.js

SofiBecerril committed:
feat: :sparkles: Generate master's and specialities charts using Node.js

SofiBecerril committed:
feat: :sparkles: Generate master's and specialities charts using Node.js

SofiBecerril committed:
Fix Prettier styling 🎨

SofiBecerril committed:
Fix Prettier styling 🎨

SofiBecerril committed:
Fix Prettier styling 🎨

SofiBecerril committed:
Merge branch 'dev-sofi' of https://github.com/OnlyAlec/ULSA_ProcesosPosgrado into dev-sofi

SofiBecerril committed:
Fix PHP styling 🎨

SofiBecerril committed:
fix: :bug: Fix chart generation on Excel report

SofiBecerril committed:
fix: :bug: Fix chart generation on Excel report

SofiBecerril committed:
Merge branch 'dev-sofi' of https://github.com/OnlyAlec/ULSA_ProcesosPosgrado into dev-sofi

SofiBecerril committed:
Fix Prettier styling 🎨

SofiBecerril committed:
Fix Prettier styling 🎨

EliBece committed:
feat: :sparkles: Add New Filters

Three new filters were added depending on the status SED of the student. Also, the logic of the icons in "Actions" was inverted.

EliBece committed:
Fix PHP styling 🎨

SofiBecerril committed:
style: :art: Chart color adjustment

SofiBecerril committed:
style: :art: Chart color adjustment

SofiBecerril committed:
Fix PHP styling 🎨

SofiBecerril committed:
Fix PHP styling 🎨

OnlyAlec commented on public/modules/AFI/formsDB.php:
```diff
@@ -255,15 +257,51 @@ function createExcel($students, $programCount)
+        }
+    }
+
+    $scriptPath = '../../assets/js/AFI/generate_chart.js';
```
Cambiemos de ubicación el JS, al ser un proceso que pasa en el servidor no es necesario que el usuario sepa donde esta el JS (Recordemos que todo lo que este en la carpeta `public` cualquiera puede verlo).

Preferiblemente que sea dentro de la carpeta de `includes`, te dejo a la libertad de crear una nueva carpeta o ahi suelto o dentro de otra subcarpeta 😀

OnlyAlec commented on public/modules/AFI/formsDB.php:
```diff
@@ -255,15 +257,51 @@ function createExcel($students, $programCount)
+    $type = escapeshellarg('especialidades');
+    $escapedGraphsDir = escapeshellarg($graphsDir);
+
+    exec("node $scriptPath $escapedTempFile $type $escapedGraphsDir");
```
En lugar de usar `exec` usemos `Process` (Symfony\Component\Process\Process). Podemos usar de inspiración el método `getLocalResult()` de la clase `Mjml` (`vendor > spatie > mjml-php > Mjml `).

Si se complica mucho,  lo dejamos así  👍🏻

OnlyAlec commented on public/modules/SED/generate_report.php:
```diff
@@ -100,12 +102,13 @@ function generateReport($students, $filename)
         foreach ($studentsByEvaluation['EVALUATED'] as $programName => $programStudents) {
             addTable($pdf, $programName, $programStudents);
             $countEvaluated = $countEvaluated + count($programStudents);
+            $countEvaluated = $countEvaluated + count($programStudents);
```
Posible duplicado

OnlyAlec commented on public/modules/SED/generate_report.php:
Cambiar nuevamente font de `Helvetica` a  `IndivisaSans`

OnlyAlec reviewed (changes requested):
Alooh, hay nuevos cambios relacionados con lo que desarrollaste y como adicional otros  relacionados con el modulo o actualización de tu PR.

OnlyAlec committed:
Merge pull request #96 from OnlyAlec/dev-eli

Add New Filters

SofiBecerril committed:
refactor: :adhesive_bandage: Duplicated deleted

SofiBecerril committed:
feat: :adhesive_bandage: Font change

SofiBecerril committed:
feat: :adhesive_bandage: Font change

SofiBecerril committed:
feat: :adhesive_bandage: Font change

SofiBecerril committed:
feat: :adhesive_bandage: Font change

SofiBecerril committed:
feat: :adhesive_bandage: Relocate generate_charts.js

SofiBecerril committed:
feat: :adhesive_bandage: Relocate generate_charts.js

SofiBecerril committed:
refactor: :recycle: Switched from exec() to Process

SofiBecerril committed:
refactor: :recycle: Switched from exec() to Process

SofiBecerril committed:
Fix PHP styling 🎨

SofiBecerril committed:
Fix PHP styling 🎨

SofiBecerril committed:
Merge branch 'dev-sofi' of https://github.com/OnlyAlec/ULSA_ProcesosPosgrado into dev-sofi

SofiBecerril committed:
refactor: :adhesive_bandage: Removed duplicated lines

SofiBecerril committed:
Fix Prettier styling 🎨
